packages/client/hmi-client/src/components/widgets/tera-timestep-calendar.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+	<div class="flex flex-column gap-2">
+		<label>{{ label }}</label>
+		<tera-input-number
+			:model-value="modelValue"
+			:disabled="disabled"
+			@update:model-value="emit('update:model-value', $event)"
+		/>
+		<Calendar
+			v-if="modelConfiguration.temporalContext"
+			:model-value="date"
+			showIcon
+			iconDisplay="input"
+			:view="calendarSettings?.view"
+			:date-format="calendarSettings?.format"
+			:disabled="disabled"
+			@date-select="onDateSelect($event)"
+		/>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { Model, ModelConfiguration } from '@/types/Types';
+import Calendar from 'primevue/calendar';
+import {
+	CalendarSettings,
+	getCalendarSettingsFromModel,
+	getEndDateFromTimestep,
+	getTimestepFromDateRange
+} from '@/utils/date';
+import { computed, ref } from 'vue';
+import teraInputNumber from './tera-input-number.vue';
+
+const props = defineProps<{
+	label: string;
+	modelValue: number;
+	model: Model;
+	modelConfiguration: ModelConfiguration;
+	disabled?: boolean;
+}>();
```

Comment:
This component feels awkwardly constructed - if your intention is to create a general timestep/calendar component it shouldn't know anything about model and model-configuration.

Widgets, like the utils folders, are originally meant to house domain agnostic items. So I would suggest we either make this generic, or to move this out of widget-folder and give it a more specific name.

Reply:
Makes sense. I've made the component more generic such that it doesn't rely on `model` or `model-configuration` anymore.  Let me know what you think!